app/main/views/frameworks.py

```diff
@@ -273,6 +274,80 @@ def framework_start_supplier_declaration(framework_slug):
 
 
 @main.route('/frameworks/<framework_slug>/declaration', methods=['GET'])
+@login_required
+def framework_supplier_declaration_overview(framework_slug):
+    framework = get_framework(data_api_client, framework_slug, allowed_statuses=['open'])
+
+    sf = data_api_client.get_supplier_framework_info(current_user.supplier_id, framework_slug)["frameworkInterest"]
+    # ensure our declaration is at least (probably) a dict
+    sf["declaration"] = sf.get("declaration") or {}
+
+    content = content_loader.get_manifest(framework_slug, 'declaration').filter(sf["declaration"])
+
+    # we must perform an actual validation for each section rather than rely on .answer_required as the latter won't
+    # take into account declarations custom question dependencies
+    sections_errors = OrderedDict(
+        (
+            section.slug,
+            (
+                section,
+                section.editable and get_validator(
+                    framework,
+                    content,
+                    sf["declaration"],
+                ).get_error_messages_for_page(section),
+            ),
+        )
+        for section in content.summary(sf["declaration"])
+    )
+
+    return render_template(
+        "frameworks/declaration_overview.html",
+        framework=framework,
+        supplier_framework=sf,
+        sections_errors=sections_errors,
+        validates=not any(errors for section, errors in sections_errors.values()),
+        framework_dates=content_loader.get_message(framework_slug, "dates"),
+    )
```

Comment:
Oh look at me the django person.

Doesn't it return 200 by default? Will add it for clarity anyway.